site/components/ui/ThemeSwitcher/ThemeSwitcher.tsx

```diff
@@ -1,18 +1,23 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import { useTheme } from 'next-themes'
-import { ChevronUp, Cross } from '@components/icons'
+import { ChevronRight, Cross } from '@components/icons'
 
 import cn from 'clsx'
 import ClickOutside from '@lib/click-outside'
 import ThemeIcon from './ThemeIcon'
 
 const ThemeSwitcher = () => {
+  const [mounted, setMounted] = useState(false)
   const [display, setDisplay] = useState(false)
   const { theme, themes, setTheme } = useTheme()
 
+  useEffect(() => setMounted(true), [])
+
+  if (!mounted) return null
+
```

Comment:
This feels wrong, something like this makes more sense to me: https://github.com/pacocoursey/next-themes/issues/153#issuecomment-1324246080